p4-16/spec/P4-16-spec.mdk

```diff
@@ -7078,17 +7084,18 @@ annotations
 annotation
     : '@' name
     | '@' name '(' annotationBody ')'
+    | '@' name '[' structuredAnnotationBody ']'
```

Comment:
Perhaps you can use the grammar from the compiler implementation ; it essentially inlines the structuredAnnotationBody in this rule.

Reply:
If this suggestion is taken, be sure to patch up those places where the text refers to `structuredAnnotationBody`.

Reply:
What is the advantage of inlining here? I think I like that we have a name for structured annotation bodies.

Reply:
The spec grammar is almost identical to the bison grammar, because then we know it works. If we deviate it may actually not be implementable. Bison is very peculiar.